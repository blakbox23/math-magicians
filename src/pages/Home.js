import React from 'react';

function Home() {
  return (
    <div className="homePg">
      <h1>Welcome to our page!</h1>
      <p data-testid="firstPar">
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
      </p>
      <br />
      <p data-testid="secondPar">
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
        Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
      </p>
    </div>
  );
}

export default Home;
