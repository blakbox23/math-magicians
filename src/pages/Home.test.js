import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from './Home';

describe('Home', () => {
  test('renders Home component', () => {
    render(<Home />);
    expect(screen.getByText('Welcome to our page!')).toBeInTheDocument();
  });
  test('renders Home component with first paragraph', () => {
    const { getByTestId } = render(<Home />);
    expect(getByTestId('firstPar')).toHaveTextContent('Donec quam');
  });
  test('renders Home component with first paragraph', () => {
    const { getByTestId } = render(<Home />);
    expect(getByTestId('secondPar')).toHaveTextContent('Nulla consequat');
  });
});

describe('Home snapshot', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
