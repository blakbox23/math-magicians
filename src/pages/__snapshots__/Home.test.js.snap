// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot renders correctly 1`] = `
<div
  className="homePg"
>
  <h1>
    Welcome to our page!
  </h1>
  <p
    data-testid="firstPar"
  >
    Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
  </p>
  <br />
  <p
    data-testid="secondPar"
  >
    Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat
  </p>
</div>
`;
